sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Max.scala

```diff
@@ -57,3 +57,34 @@ case class Max(child: Expression) extends DeclarativeAggregate {
 
   override lazy val evaluateExpression: AttributeReference = max
 }
+
+abstract class AnyAggBase(arg: Expression)
+  extends UnevaluableAggrgate with ImplicitCastInputTypes {
+
+  override def children: Seq[Expression] = arg :: Nil
+
+  override def dataType: DataType = BooleanType
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(BooleanType)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    arg.dataType match {
+      case dt if dt != BooleanType =>
+        TypeCheckResult.TypeCheckFailure(s"Input to function '$prettyName' should have been " +
+          s"${BooleanType.simpleString}, but it's [${arg.dataType.catalogString}].")
+      case _ => TypeCheckResult.TypeCheckSuccess
+    }
+  }
+}
+
+@ExpressionDescription(
+  usage = "_FUNC_(expr) - Returns true if at least one value of `expr` is true.")
```

Comment:
I will add. One observation : 

I see that, only in the API's we specify the @since .. for example for aggregate Max, we have it in api function.scala:Max ..  These aggregates are not exposed in the dataset apis and none of the other aggregates seem to have it.

Reply:
I think the point is that it was not available until 2.3, so earlier methods don't have it. Am I missing something?

Reply:
@mgaido91 Yeah... If we look at the definition of other aggregate function like Max, Min etc, they don't seem to have the `@Since`.  However, they are defined in the for `functions.scala` for max and min as `@Since 1.3`.  So basically i was not sure on the what is the rule when a function is not exposed in dataset api but only from SQL.

Reply:
just to be clear, here I am not talking about `@Since`, I am talking about `since` as parameter of `@ExpressionDescription`

Reply:
Ideally we need `Since` here. Some functions don't have them because at that time the `Since` method was not there. We should add missing `Since` to them as well, if other people have time to do it.

Reply:
@mgaido91 @cloud-fan Thanks .. will add.

Reply:
(btw, let's add `since` at `ExpressionDescription` wherever possible .. )

Reply:
@HyukjinKwon Sure.. I will open a pr shortly.